src/backend/distributed/utils/metadata_cache.c

```diff
@@ -435,28 +491,14 @@ DistributedTableCacheEntry(Oid distributedRelationId)
 {
 	DistTableCacheEntry *cacheEntry = NULL;
 
-	/*
-	 * Can't be a distributed relation if the extension hasn't been loaded
-	 * yet. As we can't do lookups in nonexistent tables, directly return NULL
-	 * here.
-	 */
-	if (!CitusHasBeenLoaded())
+	if (!IsDistributedTable(distributedRelationId))
```

Comment:
This *doubles* the lookup cost in the common case that this actually is a distributed table. Again, I don't think we should do this for the normal case.